### ISSUE TITLE
Retain "Sent notifications" after Form Import/Update

### ISSUE THREAD

**Issue by benediktblank** [verbb/formie#1047]
### Describe the bug

Whenever we import an existing form via the Import/Export functionality from Formie, all "Sent notifications" for that form are deleted from the database.

I had a look at the code and I think I found the reason in vendor/verbb/formie/src/helpers/ImportExportHelper.php (253).

It's because the forms existing notifications are not updated but newly created with each import.
That way they get a new ID and the foreign key connection between "formie_sentnotifications.notificationId" and "formie_notifications.id" removes the lines in "formie_sentnotifications" that have no matching notificationId anymore.

Is there any way to prevent this from happening?

Could there be an option to update the existing form notification rather than recreating with each import?
Maybe with adding a notification handle to identify the existing notification for updating?
That would be great.

The reason I'm asking is mainly, because we are importing our Forms with each deployment via the exported JSON files.
The exported JSON files are versioned in the repository.
We built a console command for this "import during deployment" in case you might be interested.

### Steps to reproduce

1. Create a simple form with a Notification
2. Submit something via that form
3. Look at "Sent Notifications" list in Formie, you should see the sent notification for that form there.
4. Export the form
5. Re-Import/Update the form
6. Look at "Sent Notifications" list in Formie, the sent notification is gone now.

### Form settings

- Multi-page form: No
- Submission Method: Gql
- Client-side Validation: No
- Custom Form Templates: Yes

### Craft CMS version

4.2.1.1

### Plugin version

2.0.5

### Multi-site?

No

**Comment by engram-design** [comment]
Yeah, the reason it's tricky at the moment is the lack of an identifier for a notification, that isn't an ID. We would need to introduce a [handle](https://github.com/verbb/formie/issues/1048) for notifications to be included in the export, which in turn would be looked up against any existing notifications on a form and update/add those. Otherwise, we have no real way of knowing if there's an existing notification on the form to use and update. Their IDs will likely be different across different environments.

We _could_ use the name of a notification, but that seems a little risky. We could also maybe use `uid`.

I'll give it some thought.

**Comment by benediktblank** [comment]
Yes, I also think a handle would be the best solution for this. The handle could be pre-filled from the name like you do with the Form fields. That way it can be changed if need be.

Combined with the formId in the formie_notifications table it would still be possible to have the same notification handle on different forms but not on the same form.

Thanks for adding the feature request.

**Comment by engram-design** [comment]
This should be resolved in [3.0.0](https://github.com/verbb/formie/releases/tag/3.0.0) with the introduction of handles for email notifications. We won't be looking to add this into Formie 2.
